lib/archethic_web/controllers/faucet_controller.ex

```diff
@@ -14,6 +14,19 @@ defmodule ArchEthicWeb.FaucetController do
 
   @pool_seed Application.compile_env(:archethic, __MODULE__)[:seed]
 
+  plug(:enabled)
+
+  defp enabled(conn, _) do
+    if Application.get_env(:archethic, ArchEthicWeb.FaucetController)
+       |> Keyword.get(:enabled, false) do
+      conn
+    else
+      conn
```

Comment:
I think you can use built-in error handling by Phoenix by using something like: `raise Phoenix.Router.NoRouteError, conn: conn, router: router`

Reply:
I have already looked this up but it's only available during debugging (the NoRouteError page).

Reply:
You can test this using any random route in mainnet.archethic.com.

Reply:
If you go to `https://mainnet.archethic.net/faucet` it will show `{"errors":{"detail":"Not Found"}}`.
So in dev, you will have an error page, but on the prod, you will have the JSON render

Reply:
Yes. Since I was deploying this in prod I thought I would replicate the behaviour in mainnet. So do I detect mix.env() and then output different error pages based on that?

Reply:
in the config.exs you have this:

```
config :archethic, ArchEthicWeb.Endpoint,
  secret_key_base: "5mFu4p5cPMY5Ii0HvjkLfhYZYtC0JAJofu70bzmi5x3xzFIJNlXFgIY5g8YdDPMf",
  render_errors: [view: ArchEthicWeb.ErrorView, accepts: ~w(json)],
  pubsub_server: ArchEthicWeb.PubSub,
  live_view: [
    signing_salt: "3D6jYvx3",
    layout: {ArchEthicWeb.LayoutView, "live.html"}
  ]
```

and in dev.exs you have a `debug` flag enabled, so in prod it will display json and on dev a specific UI

Reply:
You can use something like this otherwise:
```elixir
conn                                                                                         
|> put_status(:not_found)                                                                        
|> render(ArchEthicWeb.ErrorView, "404.html")
```

Reply:
I used the following.
```elixir
conn
|> put_status(:not_found)
|> put_view(ArchEthicWeb.ErrorView)
|> render("404.html")
|> halt()
```

This is the output during dev mode.
<img width="1463" alt="Screenshot 2021-09-27 at 21 16 51" src="https://user-images.githubusercontent.com/24897721/134941682-80313c27-b516-436f-863b-6b5e2336cf8f.png">

Reply:
@samuel-uniris Is this the desired behaviour? This is showing in both dev and prod.